### PR TITLE
Add timestamp formatting to EventStringFormatter

### PR DESCRIPTION
**Pull request by urso**
Add date formatter to libbeat/common/dtfmt. Pattern syntax supported
is subset of joda DateTimeFormat

### CODE REVIEW

**Comment by ruflin** [comment]
We will need this for `index_pattern`?

**Comment by urso** [comment]
yes. I want similar formatting support as logstash for `index_pattern`

File: libbeat/common/dtfmt/dtfmt_test.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+		{mkDate(2006, 8, 1), "YYYY.MMMM.d", "2006.August.1"},
+		{mkDate(2006, 8, 1), "YYYYYY.MM.ddd", "002006.08.001"},
+
+		// week year + week of year + day of week
+		{mkDate(2015, 1, 1), "xx.ww.e", "15.01.4"},
+		{mkDate(2014, 12, 31), "xx.ww.e", "15.01.3"},
+		{mkDate(2015, 1, 1), "xx.w.E", "15.1.Thu"},
+		{mkDate(2014, 12, 31), "xx.w.E", "15.1.Wed"},
+		{mkDate(2015, 1, 1), "xx.w.EEEE", "15.1.Thursday"},
+		{mkDate(2014, 12, 31), "xx.w.EEEE", "15.1.Wednesday"},
```
Comment (tsg) [inline libbeat/common/dtfmt/dtfmt_test.go]: Maybe also add a test with `xxxx.ww` which is the format recommended by the Logstash docs.

Comment (urso) [inline libbeat/common/dtfmt/dtfmt_test.go]: done

**Comment by tsg** [comment]
Amazing work, @urso! This code is actually a lot cleaner than the stdlib formatter. 👍 

LGTM

**Pull request merged**